packages/xchain-util/src/async.test.ts

```diff
@@ -0,0 +1,65 @@
+import nock from 'nock'
+// eslint-disable-next-line ordered-imports/ordered-imports
+import { getAllInboundDetails, getInboundDetails, getMimirDetails } from './async'
+import { Chain } from './chain'
+
+describe('functions from Midgard', () => {
+  it('should return mimir details', async function () {
+    const expectedResponse = { testProperty: 'test value' }
+    nock('https://midgard.ninerealms.com').get('/v2/thorchain/mimir').reply(404)
+    nock('https://midgard.thorswap.net').get('/v2/thorchain/mimir').reply(200, expectedResponse)
+
+    await expect(getMimirDetails()).resolves.toEqual(expectedResponse)
+
+    nock.restore()
+  })
+
+  it('should return inbound details for all', async function () {
```

Comment:
Pls add another tests for `getAllInboundDetails` if one or both Midgards are failing (404) - similar to previous test of `getMimirDetails`